types/aws-kcl/index.d.ts

```diff
@@ -0,0 +1,133 @@
+     *          once the initialization operation is complete.
+     */
+    initialize(initializeInput: InitializeInput, completeCallback: Callback): void;
+
+    /**
+     * Called by KCL with a list of records to be processed and checkpointed.
+     * A record looks like:
+     *
+     * The checkpointer can optionally be used to checkpoint a particular sequence
+     * number (from a record). If checkpointing, the checkpoint must always be
+     * invoked before calling `completeCallback` for processRecords. Moreover,
+     * `completeCallback` should only be invoked once the checkpoint operation
+     * callback is received.
+     *
+     * @param processRecordsInput       Process records information with
+     *             array of records that are to be processed.
+     * @param completeCallback          The callback that must be invoked
+     *             once all records are processed and checkpoint (optional) is
+     *             complete.
+     */
+    processRecords(processRecordsInput: ProcessRecordsInput, completeCallback: Callback): void;
+
+    /**
+     * Called by the KCL to indicate that this record processor should shut down.
+     * After the lease lost operation is complete, there will not be any more calls to
+     * any other functions of this record processor. Clients should not attempt to
+     * checkpoint because the lease has been lost by this Worker.
+     *
+     * @param leaseLostInput    Lease lost information.
+     * @param completeCallback  The callback must be invoked once lease
+     *             lost operations are completed.
+     */
+    leaseLost(leaseLostInput: LeaseLossInput, completeCallback: Callback): void;
+    /**
+     * Called by the KCL to indicate that this record processor should shutdown.
+     * After the shard ended operation is complete, there will not be any more calls to
+     * any other functions of this record processor. Clients are required to checkpoint
+     * at this time. This indicates that the current record processor has finished
+     * processing and new record processors for the children will be created.
+     *
+     * @param shardEndedInput       ShardEnded information.
+     * @param completeCallback      The callback must be invoked once shard
+     *               ended operations are completed.
+     */
+    shardEnded(shardEndedInput: ShardEndedInput, completeCallback: Callback): void;
+}
+
+interface KCLInput {
+    recordProcessor: RecordProcessor;
+    version: symbol;
+}
+
+export class KCLManager {
+    constructor(
+        kclManagerInput: KCLInput,
+        inputFile?: fs.ReadStream,
+        outputFile?: fs.WriteStream,
+        errorFile?: fs.WriteStream,
+    );
+    checkpoint(sequenceNumber: string): void;
+    run(): void;
+}
+
+declare function KCLProcess(
+    recordProcessor: RecordProcessor,
+    inputFile?: fs.ReadStream,
+    outputFile?: fs.WriteStream,
+    errorFile?: fs.WriteStream,
+): KCLManager;
+
+export default KCLProcess;
```

Comment:
It seems like the [underlying library](https://unpkg.com/aws-kcl@2.1.0/index.js) is not using `export default`, so this .d.ts file should not either. Instead, you should follow the appropriate template from [the TypeScript handbook](http://www.typescriptlang.org/docs/handbook/declaration-files/templates.html). Note that if you want to use `import name from 'package';` in your consuming code, you can use the `esModuleInterop` compiler setting.